src/styles/main.scss

```diff
@@ -0,0 +1,217 @@
+  height: 100vh;
+}
+
+.calendar {
+  gap: 1px;
+  display: flex;
+  flex-wrap: wrap;
+  width: 707px;
+  margin: auto;
+}
+
+.calendar:nth-child(7n) {
+  margin-top: 0;
+  padding-left: 1px;
+}
+
+.calendar .calendar__day {
+  width: 100px;
+  height: 100px;
+  background-color: #eee;
+  border: 1px solid #000;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  text-align: center;
+  transition: all 0.5s;
+  cursor: pointer;
+  padding-left: 1px;
+}
+
+.calendar .calendar__day:nth-child(1)::before {
+  content: '1';
+}
+
+.calendar .calendar__day:nth-child(2)::before {
+  content: '2';
+}
+
+.calendar .calendar__day:nth-child(3)::before {
+  content: '3';
+}
+
+.calendar .calendar__day:nth-child(4)::before {
+  content: '4';
+}
+
+.calendar .calendar__day:nth-child(5)::before {
+  content: '5';
+}
+
+.calendar .calendar__day:nth-child(6)::before {
+  content: '6';
+}
+
+.calendar .calendar__day:nth-child(7)::before {
+  content: '7';
+}
+
+.calendar .calendar__day:nth-child(8)::before {
+  content: '8';
+}
+
+.calendar .calendar__day:nth-child(9)::before {
+  content: '9';
+}
+
+.calendar .calendar__day:nth-child(10)::before {
+  content: '10';
+}
+
+.calendar .calendar__day:nth-child(11)::before {
+  content: '11';
+}
+
+.calendar .calendar__day:nth-child(12)::before {
+  content: '12';
+}
+
+.calendar .calendar__day:nth-child(13)::before {
+  content: '13';
+}
+
+.calendar .calendar__day:nth-child(14)::before {
+  content: '14';
+}
+
+.calendar .calendar__day:nth-child(15)::before {
+  content: '15';
+}
+
+.calendar .calendar__day:nth-child(16)::before {
+  content: '16';
+}
+
+.calendar .calendar__day:nth-child(17)::before {
+  content: '17';
+}
+
+.calendar .calendar__day:nth-child(18)::before {
+  content: '18';
+}
+
+.calendar .calendar__day:nth-child(19)::before {
+  content: '19';
+}
+
+.calendar .calendar__day:nth-child(20)::before {
+  content: '20';
+}
+
+.calendar .calendar__day:nth-child(21)::before {
+  content: '21';
+}
+
+.calendar .calendar__day:nth-child(22)::before {
+  content: '22';
+}
+
+.calendar .calendar__day:nth-child(23)::before {
+  content: '23';
+}
+
+.calendar .calendar__day:nth-child(24)::before {
+  content: '24';
+}
+
+.calendar .calendar__day:nth-child(25)::before {
+  content: '25';
+}
+
+.calendar .calendar__day:nth-child(26)::before {
+  content: '26';
+}
+
+.calendar .calendar__day:nth-child(27)::before {
+  content: '27';
+}
+
+.calendar .calendar__day:nth-child(28)::before {
+  content: '28';
+}
+
+.calendar .calendar__day:nth-child(29)::before {
+  content: '29';
+}
+
+.calendar .calendar__day:nth-child(30)::before {
+  content: '30';
+}
+
+.calendar .calendar__day:nth-child(31)::before {
+  content: '31';
+}
+
+.calendar .calendar__day:hover {
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+  cursor: pointer;
+}
+
+.calendar--start-day-mon > :first-child {
+  margin-left: 0;
+}
+
+.calendar--start-day-tue > :first-child {
+  margin-left: 101px;
+}
```

Comment:
Need to use @each for add margin left
https://sass-lang.com/documentation/at-rules/control/each/